runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -4221,6 +4229,12 @@ J9::Z::CodeGenerator::inlineDirectCall(
          }
       }
 
+   if (cg->getSupportsInlineStringIndexOf())
+      {
+      resultReg = inlineFindElementFromArray(node, cg);
```

Comment:
This looks incorrect. Shouldn't we be checking what the recognized method is before blindly inlining? Won't this fail try to inline every recognized method that comes to this point? For example what happens if you have a `java.lang.Math.min` call? Woudn't we enter the above `if` when we're not supposed to?

Reply:
70e2b81